src/core/public/application/application_service.tsx

```diff
@@ -517,6 +517,9 @@ export class ApplicationService {
       if (!confirmed) {
         return false;
       }
+      if (action?.callback) {
+        action.callback();
```

Comment:
```suggestion
        action.callback(confirmed);
```
We should pass the `confirmed` to the callback.

Reply:
Actually, we do not need to pass the `confirmed`, because if user clicks the cancel button, the callback function will not be executed.

Reply:
Oh, I see.